source/navier_stokes_matrix.cc

```diff
@@ -852,7 +863,10 @@ NavierStokesMatrix<dim>::local_operation(
                 grad_u[d][d] -= pres;
             }
 
-          velocity.submit_gradient(grad_u, q);
+          if constexpr (dim == 1)
+            velocity.submit_gradient(grad_u[0], q);
+          else
+            velocity.submit_gradient(grad_u, q);
```

Comment:
Wouldn't this suggest that we should another `FEEvaluationAccess<1,1,...>::submit_gradient(const Tensor<2,1,..> &)` function to accept a rank-2 tensor in deal.II? If you agree, I would support adding that function in deal.II.

Reply:
That's a good idea! @mschreter would you be willing to open a PR to enable `submit_gradient()` and `get_divergence()` for 1D? If you have questions, feel free to reach out! If you don't have time or it is too hard, I can do it later today or tomorrow.

Reply:
Thank you, I agree that it would be nice to have it already available in deal.II. And yes, it would be my pleasure to open a PR :-). @peterrum: I will propose the PR in the evening.

Reply:
I've opened PR [#dealii/11952](https://github.com/dealii/dealii/pull/11952) to solve this issue. If you have any improvements on that, I would be happy to include them.